rfcs/text/0030/file_description.yml

```diff
@@ -0,0 +1,178 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: fd
+  title: File description
+  group: 2
+  short: These fields contain information about a file descriptor.
+  description: >
+    These fields contain information about a file descriptor.
+
+    In Unix and Unix-like computer operating systems, a file descriptor (FD, less frequently fildes) is a unique identifier (handle) for a file or other input/output resource, such as a pipe or network socket.
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - at: process
+        as: tty
+        short_override: The file descriptor for the controlling tty (ctty).
+      - at: output
+        as: fd
+        short_override: The file descriptor producing the output.
+
+  fields:
+    - name: type
+      level: extended
+      type: keyword
+      short: The type of file descriptor.
+      description: >
+        The type of file descriptor.
+
+        One of: file, char_device, pipe, fifo, socket, epoll, signal, inotify, bpf, event, perf_event, timer, userfault, namespace, io_uring
+      example: char_device
+
+    - name: descriptor
+      level: extended
+      type: short
+      short: The integer representation of the FD.
```

Comment:
to be safe this should be an unsigned long (what Linux uses for a file descriptor).   E.g. a web server may have a ton of FDs - one for each connection.